docs/designers-developers/developers/tutorials/javascript/js-build-setup.md

```diff
@@ -0,0 +1,236 @@
+# JavaScript Build Setup
+
+This page covers how to set up your development environment to use the ESNext syntax. We call ESNext to JavaScript code written using features that are only available in a specification greater than ECMAScript 5 (ES5 for short) or that includes custom syntax such as [JSX](https://reactjs.org/docs/introducing-jsx.html).
+
+This documentation covers development for your plugin to work with Gutenberg, to setup a development environment to work directly on Gutenberg, see the [CONTRIBUTING.md](https://github.com/WordPress/gutenberg/blob/master/CONTRIBUTING.md) documentation.
+
+For simpler code, you can write JavaScript using ESNext and JSX syntax. This requires a transformation step to convert that syntax which a web browser may not understand, to one that it will.  Webpack and babel are the tools that perform this transformation step.
+
+[Webpack](https://webpack.js.org/) is a pluggable tool that processes JavaScript creating a compiled bundle that runs in a browser. [Babel](https://babeljs.io/) transforms JavaScript from one format to another. You use Babel as a plugin to Webpack to transform both ESNext and JSX to JavaScript.
+
+## Quick Start
+
+For a quick start, you can use one of the examples from the [Gutenberg Examples repository](https://github.com/wordpress/gutenberg-examples/). Each one of the `-esnext` directories contain the necessary files for working with ESNext and JSX.
+
+## Setup
+
+Both Webpack and Babel are tools written in JavaScript and run using [Node.js](https://nodejs.org/). Node.js is a runtime environment for JavaScript outside of a browser. Simply put, node allows you to run JavaScript code on the command-line.
+
+First, you need to set up node for your development environment. The steps required change depending on your operating system, but if you have a package manager installed, setup can be as straightforward as:
+
+- Ubuntu: `apt install node`
+- macOS: `brew install node`
+- Windows: `choco install node`
+
+Additionally, the [Node.js download page](https://nodejs.org/en/download/) includes installers and binaries.
+
+**Note:** The build tools and process occur on the command-line, so some basic familiarity using a terminal application is required. Some text editors have a terminal built-in which is fine to use, Visual Studio Code editor is one popular example.
+
+### Node Package Manager (npm)
+
+The Node Package Manager (npm) is a tool included with node. npm allows you to install and manage JavaScript packages. npm can also generate and process a special file called `package.json`, which contains some information about your project and the packages your project uses.
+
+To start a new node project, first create a directory to work in.
+
+```
+mkdir myguten-block
+```
+
+You create a new package.json using `npm init`.  This will walk you through creating your package.json file:
+
+```
+$ npm init
```

Comment:
I'm fine with the leading dollar sign to represent the system's terminal prompt, but if we choose it then:

* It must be used consistently in all snippets concerning system commands.
* It should be introduced as a documentation concept, lest a user type the symbol in.

Reply:
I removed the `$` and added a brief explanation that it is run in terminal. The `$` wasn't used elsewhere on page, so for consistency makes sense to remove.